app/controllers/timecodes_controller.rb

```diff
@@ -0,0 +1,58 @@
+class TimecodesController < ApplicationController
+  before_action :set_timecode, only: %i[edit update show destroy send_timecode]
+  before_action :set_episode, only: %i[index new create]
+
+  load_and_authorize_resource
+
+  def index; end
+
+  def show; end
+
+  def new
+    @timecode = @episode.timecodes.new
+  end
+
+  def create
+    timecode = @episode.timecodes.create(timecode_params)
+
+    if timecode.valid?
+      redirect_to episode_timecodes_path
+    else
+      redirect_to new_episode_timecode_path
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @timecode.update(timecode_params)
```

Comment:
This is about all controllers but look into this, please, and ask some questions.

https://medium.com/rails-ember-beyond/error-handling-in-rails-the-modular-way-9afcddd2fe1b
https://www.thegreatcodeadventure.com/rails-api-painless-error-handling-and-rendering-2/
https://blog.bigbinary.com/2016/02/29/rails-5-improves-redirect_to_back-with-redirect-back.html

Feel free to resolve this one for now